modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/function/EqualsFunction.java

```diff
@@ -35,7 +36,43 @@
 
 	@Override
 	public Boolean apply(Object object1, Object object2) {
-		return Objects.equals(_getValue(object1), _getValue(object2));
+		Object value1 = _getValue(object1);
+
+		Object value2 = _getValue(object2);
+
+		Object object1Class = value1.getClass();
+
+		Object object2Class = value2.getClass();
+
+		if (!object1Class.equals(object2Class) &&
+			((object1 instanceof BigDecimal) ||
+			 (object2 instanceof BigDecimal))) {
+
+			if (object1 instanceof BigDecimal) {
+				if (value2 instanceof Long) {
+					value1 = GetterUtil.getLong(value1);
+				}
+				else if (value2 instanceof Double) {
+					value1 = GetterUtil.getDouble(value1);
+				}
+				else if (value2 instanceof Integer) {
+					value1 = GetterUtil.getInteger(value1);
+				}
+			}
+			else {
+				if (value1 instanceof Long) {
+					value2 = GetterUtil.getLong(value2);
+				}
+				else if (value1 instanceof Double) {
+					value2 = GetterUtil.getDouble(value2);
+				}
+				else if (value1 instanceof Integer) {
+					value2 = GetterUtil.getInteger(value2);
+				}
+			}
+		}
+
+		return Objects.equals(value1, value2);
```

Comment:
Hi @sontruongces,
We can simplify this code by checking if both values ​​are instances of `Number`. If they are, we can convert them to `BigDecimal` as follows:

```
@Override
public Boolean apply(Object object1, Object object2) {
	if ((object1 instanceof Number) && (object2 instanceof Number)) {
		BigDecimal bigDecimal1 = new BigDecimal(object1.toString());
		BigDecimal bigDecimal2 = new BigDecimal(object2.toString());

		return bigDecimal1.compareTo(bigDecimal2) == 0;
	}

	return Objects.equals(_getValue(object1), _getValue(object2));
}
```